### DISCUSSION TITLE
WaterPressure AeroTerm HP.

### DISCUSSION THREAD

**Discussion by ProdanAS** [john30/ebusd#1189] [General]
I've setup the Ebusd adaptor and publish to MQTT, this works flawless, I get a lot of information, but water pressure is Unknown. 

I can't seem to get if this has not yet been decoded, or if im missing something, can anyone help me along ? 

version: ebusd 23.3.23.3                                                                                                                            
update check: OK                                                                                                                                    
device: 10.10.10.180:9999, TCP, enhanced, firmware 1.1[4106].1[4106]                                                                                
signal: acquired                                                                                                                                    
symbol rate: 60                                                                                                                                     
max symbol rate: 124                                                                                                                                
min arbitration micros: 1                                                                                                                           
max arbitration micros: 29                                                                                                                          
min symbol latency: 7                                                                                                                               
max symbol latency: 44                                                                                                                              
scan: finished                                                                                                                                      
reconnects: 0                                                                                                                                       
masters: 3                                                                                                                                          
messages: 69                                                                                                                                        
conditional: 0                                                                                                                                      
poll: 25                                                                                                                                            
update: 10                                                                                                                                          
address 03: master #11                                                                                                                              
address 08: slave #11, scanned "MF=Vaillant;ID=HMU00;SW=0607;HW=5103", loaded "vaillant/08.hmu.csv"                                                 
address 31: master #8, ebusd                                                                                                                        
address 36: slave #8, ebusd                                                                                                                         
address 71: master #9                                                                                                                               
address 76: slave #9, scanned "MF=Vaillant;ID=VWZIO;SW=0202;HW=0103"                                                                                
                                                                          

<img width="325" alt="Skærmbillede 2024-02-07 kl  15 28 02" src="https://github.com/john30/ebusd/assets/116255260/025a66ad-cf86-4647-948f-7ddf2a36afb7">

**Comment by rafapascual** [comment]
Hi

similar or same issue here (I'm not using HA, I will integrate with openhab trough MQTT if possible). Seems like the value is on a different location than defined in the csv config file but I'm new with ebusd and I'm starting to learn how to debug and adjust it. Some light will be really appreciate.

In my case the HW is the same and SW looks a different version: 

`"MF=Vaillant;ID=HMU00;SW=0902;HW=5103", loaded "vaillant/08.hmu.csv"`

My info:

```
version: ebusd 23.3.23.3
update check: OK
device: 192.168.3.7:9999, TCP, enhanced, firmware 1.1[4106].1[4106]
signal: acquired
symbol rate: 62
max symbol rate: 179
min arbitration micros: 6
max arbitration micros: 17
min symbol latency: 8
max symbol latency: 38
scan: finished
reconnects: 0
masters: 5
messages: 635
conditional: 0
poll: 0
update: 10
address 00: master #1
address 03: master #11
address 05: slave #1, scanned "MF=Vaillant;ID=VR921;SW=2801;HW=5703"
address 08: slave #11, scanned "MF=Vaillant;ID=HMU00;SW=0902;HW=5103", loaded "vaillant/08.hmu.csv"
address 10: master #2
address 15: slave #2, scanned "MF=Vaillant;ID=BASV2;SW=0562;HW=6304", loaded "vaillant/15.basv.csv"
address 31: master #8, ebusd
address 36: slave #8, ebusd
address 71: master #9
address 76: slave #9, scanned "MF=Vaillant;ID=VWZIO;SW=0202;HW=0103"
```


I'm loading the config files from https://cfg.ebusd.eu/

I tried "read -f WaterPressure" and always return "-"

```
read -f WaterPressure
-
```

I also tried to read "BuildingCircuitWaterPressure" and always returns 0.0

```
read -f BuildingCircuitWaterPressure
0.0

```


I do not see any errors on the logs.



Thank you in advance.

**Reply by rafapascual** [reply]
Hi

I found where the data is being sent but I cannot figure out how to extract it and adjust the config file accordingly

Using the raw log I found this:

time | message | field | byte 1 | byte 2
-- | -- | -- | -- | --
2024-02-18   21:15:16.084 < | 0376b5120613 | 00**0d**4d07005d000200ffd300 | 00 | 13
2024-02-18 21:15:55.944 < | 0376b5120613 | 00**0e**63070016000200ffd300 | 00 | 14
2024-02-18 21:16:43.744 < | 0376b5120613 | 00**0f**6a0700ee000200ffd300 | 00 | 15
2024-02-18 21:17:40.123 < | 0376b5120613 | 00**0f**550700a901000200ffd300 | 00 | 15
2024-02-18 21:17:47.559 < | 0376b5120613 | 00**0e**63070016000200ffd300 | 00 | 14
2024-02-18 21:19:39.154 < | 0376b5120613 | 00**0f**6f0700a0000200ffd300 | 00 | 15
2024-02-18 21:19:47.100 < | 0376b5120613 | 00**10**73070055000200ffd300 | 00 | 16
2024-02-18 21:19:55.074 < | 0376b5120613 | 00**11**6807007a000200ffd300 | 00 | 17
2024-02-18 21:20:03.065 < | 0376b5120613 | 00**10**7007006f000200ffd300 | 00 | 16
2024-02-18 21:20:26.984 < | 0376b5120613 | 00**0f**61070064000200ffd300 | 00 | 15
2024-02-18 21:20:34.955 < | 0376b5120613 | 00**0e**6b0700a6000200ffd300 | 00 | 14


The second byte contains the Building Circuit Water Pressure (I think it is the correct field name rather than just Water Pressure, at least it fits better the description that is shown on the control display).

So 13, 14, 15, 16.... actually are 1.3, 1.4, 1.5, 1.6 bars...

There is a new message every 8 seconds, I copied and paste a few messages when there was a change on the pressure, or the message length was different.



Thanks

**Reply by rafapascual** [reply]
Hi


I found a partial solution to my issue that maybe can help @ProdanAS , the last part of this message.


Not sure if I'm missing something, because the `grab result` is giving me this output:

```
0376b5120613000e5b0700 / 0200ff = 5
0376b5120613000e5a0700 / 0200ff = 2
0376b5120613000e590700 / 0200ff = 4
0376b5120613000e580700 / 0200ff = 3
0376b5120613000e5f0700 / 0200ff = 2
```

No idea if I can extract the data from these messages, as the part after the slash is always 0200ff and the data I want is on the message part rather than the fields part?

Anyway, I also found that apart from the Water Pressure value, the message also contains the Circuit Flow rate value.

`0376b5120613000e5f0700 `
where
  - 0e = Pressure = 14 => divided by 10 = 1.4 bar
  - 5f07 =  Flow, low byte first = 1887 l/h



And after some research I found that it seems the official configuration repository does not fully work with my hmu unit, and I saw there are many people publishing modifications.

I've tested some of them until I found one that is providing a lot of values that seems reliable.

But, the Circuit Water Pressure does not work either.... (I am very interested in this value because my outdoor unit is leaking water through the security valve and Vaillant says it is a problem on the installation, but that does not seem true because there is not over pressure ever, they replaced the security valve 3 times... and the issue continues, I need to found if there is a pattern to start leaking)


Going again to the `grab result` output I also found this:

```
1008b5110100 / 094d0310000808000000 = 17
1008b5110100 / 0921030e4f0409000021 = 12
```

The 4th byte is the Water Pressure

I modified my 08.hmu.csv file to include this under the "B511" PBSB:

`r,,BuildingCircuitWaterPressure,,,,,00,,,IGN:2,,,,,,UCH,10,bar,,,,,,,,,,,,,`

```
# type (r[1-9];w;u),circuit,name,[comment],[QQ],ZZ,PBSB,[ID],field1,part (m/s),datatypes/templates,divider/values,unit,comment,field2,part (m/s),datatypes/templates,divider/values,unit,comment,field3,part (m/s),datatypes/templates,divider/values,unit,comment,field4,part (m/s),datatypes/templates,divider/values,unit,comment
*r,,,,,,"B511",,,,,,,,,,,,,,,,,,,,,,,,,
r,,State,,,,,07,energy,,UCH,,,,,,energy,,kWh,,onoff,,UCH,0=off;1=on,,,state,,UCH,0x01=ready;0x0b=error;0x09=heating;0x11=cooling;0x81=heating_water,,
r,,BuildingCircuitWaterPressure,,,,,00,,,IGN:2,,,,,,UCH,10,bar,,,,,,,,,,,,,
```



Regards
